staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -570,6 +573,9 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 				return nil, nil, err
 			} else {
 				rest.FillObjectMetaSystemFields(objectMeta)
+				if len(objectMeta.GetGenerateName()) > 0 && len(objectMeta.GetName()) == 0 {
```

Comment:
is it even possible for name to be len 0 here? update requires the name be pre-specified... populating name from generateName in update does not seem correct to me

Reply:
... this area of code is not my thing as you can see :smile:

Reply:
Yeah, update supports creating but you have to know the name.

Reply:
```
kubectl -v7 apply -f pod.yaml                           
I0830 22:49:24.561890 4007126 loader.go:372] Config loaded from file:  /tmp/kind.conf
I0830 22:49:24.562775 4007126 round_trippers.go:463] GET https://127.0.0.1:35399/openapi/v2?timeout=32s
I0830 22:49:24.562784 4007126 round_trippers.go:469] Request Headers:
I0830 22:49:24.562802 4007126 round_trippers.go:473]     Accept: application/com.github.proto-openapi.spec.v2@v1.0+protobuf
I0830 22:49:24.562808 4007126 round_trippers.go:473]     User-Agent: kubectl/v1.23.4 (linux/amd64) kubernetes/e6c093d
I0830 22:49:24.598770 4007126 round_trippers.go:574] Response Status: 200 OK in 35 milliseconds
F0830 22:49:24.738982 4007126 helpers.go:118] error: from gen-: cannot use generate name with apply
```

Reply:
hmm, this test fails, but I don't know if there is some check on the name before calling Update

```diff
+++ b/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store_test.go
@@ -1070,6 +1070,34 @@ func TestStoreCreateOnUpdateHooks(t *testing.T) {
        }
 }
 
+func TestCreateOnUpdateGenerateName(t *testing.T) {
+       pod := &example.Pod{
+               ObjectMeta: metav1.ObjectMeta{GenerateName: "foo-", Namespace: "test"},
+               Spec:       example.PodSpec{NodeName: "machine"},
+       }
+
+       testContext := genericapirequest.WithNamespace(genericapirequest.NewContext(), "test")
+       destroyFunc, registry := NewTestGenericStoreRegistry(t)
+       defer destroyFunc()
+
+       obj, ok, err := registry.Update(testContext, "", rest.DefaultUpdatedObjectInfo(pod), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, true, &metav1.UpdateOptions{})
+       if err != nil {
+               t.Fatalf("Unexpected error: %v", err)
+       }
+       if !ok {
+               t.Fatalf("Expected object to be created")
+       }
+
+       accessor, err := meta.Accessor(obj)
+       if err != nil {
+               t.Errorf("Unexpected error: %v", err)
+       }
+
+       if !strings.HasPrefix(accessor.GetName(), "foo-") {
+               t.Fatalf("Expected object name to be generated from prefix foo-")
+       }
+}
+
```

Reply:
ic, that is only valid for the test, you have to specify the name as part of the URL to call an Update